### ISSUE TITLE
About the opt training of the model 

### ISSUE THREAD

**Issue by sry-make** [JinshuaiBai/LSWR_loss_function_PINN#8]
Hello Dr.Bai:
    I have a problem with your open source code 03_perforated_plate, the parameter setting of L_BFGS_B is factr=10, pgtol=1e-10, m=50, maxls=50, maxiter=20000 the result obtained from training is the number of iterations 100-200 The loss function value is 1e-3~1e-4, which is far from the data in out.mat. I have modified the optimizer parameters several times and still no change.A Chinese graduate student sincerely seeking advice

**Comment by JinshuaiBai** [comment]
> Hello Dr.Bai: I have a problem with your open source code 03_perforated_plate, the parameter setting of L_BFGS_B is factr=10, pgtol=1e-10, m=50, maxls=50, maxiter=20000 the result obtained from training is the number of iterations 100-200 The loss function value is 1e-3~1e-4, which is far from the data in out.mat. I have modified the optimizer parameters several times and still no change.A Chinese graduate student sincerely seeking advice

Dear sry-make,

Thank you for your question. I will suggest you to try the following things:
1. You may normalise the material Young's modulus and the traction force. As you can see, the activation function we used in the neural network are normally tanh, which is a function that maps any input within the range [-1, 1]. In other words, the neural network is way more effective in terms of accuracy and efficiency if you can make sure the output is also within the range of [-1, 1]. That doesn't mean that NN cannot handle scenarios where the outputs are outside this range. If you do so (not pre-normalise your problem), the last layer will take most of the burden to do the approximation.
2. You may use ADAM optimiser first and then apply L-BFGS-B. The ADAM is more stable and memory-friendly, while the L-BFGS-B is very efficient in converging if the weights and biases in your NN are very close to the local optima. However, when you randomly initialised your NN, weights and biases are normally not around the local optima, so I will suggest you to use a more stable optimiser to let them get close to a local optima, and then apply the power L-BFGS-B (a quasi-Newton solver) to solve the problem. You can find the ADAM optimiser in my other repositories.

Hope these can give you some clues to solve your problem!

**Comment by sry-make** [comment]
Dear Dr.Bai,
      Thank you for your kind help, with your help I managed to train with good results. I will continue to study hard in this field and I wish you all the best in your research!

**Comment by JinshuaiBai** [comment]
> Dear Dr.Bai, Thank you for your kind help, with your help I managed to train with good results. I will continue to study hard in this field and I wish you all the best in your research!

Glad to hear that!

Good luck and best wishes to you!

**Comment by sry-make** [comment]
Dear Dr.Bai,Sorry to bother you again.I want to redo the 3D case in your paper. Would you please send me the code? My email's address is 1258563930@qq.com.Thanks again for your help.

**Comment by JinshuaiBai** [comment]
> sry-make

Sent. Please check it in your email.
